server/notebooks/tasks.py

```diff
@@ -22,21 +22,41 @@ def execute_notebook_revisions_cleanup(notebook_id, now_utc=None):
       (also called Tumbling windows).
     """
     draft_revisions = NotebookRevision.objects.filter(notebook_id=notebook_id, is_draft=True)
+    try:
+        latest_non_draft_revision = NotebookRevision.objects.filter(
+            notebook_id=notebook_id, is_draft=False
+        ).latest("created")
+    except NotebookRevision.DoesNotExist:
+        latest_non_draft_revision = None
```

Comment:
I don't think this should ever happen. In theory it might currently be possible to delete all the revisions of a notebook from the user page, but I would expect other weird things to happen after that. I would suggest adding something like this (you will need to set up logging inside this file, look at `server/files/tasks.py` for an example)

```suggestion
        logger.warning("Attempted to prune revisions from notebook %s with no revisions", notebook_id)
        return  # should never happen
```

Reply:
This can happen, e.g.: when you create a new notebook, there is only one draft revision and there is no non-draft revision.

This logic here is to find the latest **non-draft** revision.

Reply:
Ah right, sorry, you are correct. This makes perfect sense.